### PR TITLE
[WIP] Update dependencies and sbt build

### PR DESCRIPTION
**Pull request by easel**
I'd like to get a cats 1.0/circe0.9 compatible build published and went ahead and bumped the rest of the build dependencies in the process. If we end up with a Mill build instead of sbt, most of this will be obviated and we can just close out the PR, but I've outlined the blockers to getting to SBT 1.0 for reference.

- [X] faithful needs to be ported to cats 1.0+ (merged to master)
- [X] faithful needs to be published for cats 1.0+
- [x] sbt-ornate needs to be ported to sbt 1.0 (https://github.com/szeiger/ornate/pull/15)
- [ ] sbt-ornate needs to be published for 1.0
- [x] File.*** references in AssetsTask need to be fixed
- [x] sbt-doge usage needs to be investigated
- [x] sbt-unidoc key references need to be resolved to allow update to 0.4.1
- [ ] Make sure nothing broke

### CODE REVIEW

**Comment by julienrf** [comment]
It looks to me that we might be blocked by ornate and unidoc. The other points seem easy to fix.

**Comment by Krever** [comment]
I think we could quite easily switch to [sbt-microsites](https://github.com/47deg/sbt-microsites) or [sbt-site](https://github.com/sbt/sbt-site) if necessary.

**Comment by easel** [comment]
There's an SBT 1.0 build of ornate at https://github.com/szeiger/ornate/pull/15, but it seems stalled. I pulled it down and got it to compile locally, but didn't have much luck getting a cross compile to work.

**Comment by easel** [comment]
I *think* I've got everything ported and working. Right now we're blocked on getting a new version of sbt-ornate published for sbt1.0. I've bumped the ticket with the sbt 1.0 port, hopefully we can get that one cleared shortly, making it easier for folks to double-check that I haven't broken anything.

**Comment by easel** [comment]
@Krever yes that's correct, modulo my missing something or otherwise screwing it up.

**Comment by julienrf** [comment]
@easel I’ve just publish faithful 1.1.0 (which depends on cats 1.x) :)

**Comment by easel** [comment]
Thanks @julienrf, I've updated the PR and checklist. Also bumped sbt to 1.1.5

File: documentation/build.sbt [hunk 1]
```diff
@@ -25,11 +25,10 @@ val `json-schema-jvm` = LocalProject("json-schemaJVM")
 val `json-schema-circe-jvm` = LocalProject("json-schema-circeJVM")
 val `json-schema-generic-jvm` = LocalProject("json-schema-genericJVM")
 
-import sbtunidoc.Plugin.UnidocKeys.unidoc
-
 val apiDoc =
   project.in(file("api-doc"))
-    .settings(noPublishSettings ++ `scala 2.11` ++ unidocSettings: _*)
+    .enablePlugins(ScalaUnidocPlugin)
+    .settings(noPublishSettings ++ `scala 2.11`: _*)
```
Comment (julienrf) [inline documentation/build.sbt]: We should try to use Scala 2.12 to generate the API documentation to benefit from the new design/layout.

**Comment by julienrf** [comment]
Ornate now supports sbt 1.x! https://szeiger.github.io/ornate-doc/#v0.6

**Comment by julienrf** [comment]
Thanks @easel for your work! I’ve re-applied your changes in #333.

**Pull request closed without merging**